### PR TITLE
MC-40719: Product Thumbnails not switching properly for Config product

### PR DESCRIPTION
**Pull request by thiaramus**
## Purpose of this pull request

This pull request (PR) is to clarify the default behavior of the Image Gallery for configurable products when a user switches between product configurations. To add a note that this behavior can be controlled with the swatch attribute settings, and by the theme.

Over the years we had many opened support tickets due to the misunderstanding of the default behavior since it wasn't described anywhere. 

## Magento release version

- [x] 2.4.x

   Specify a patch release number, if applicable:

- [x] 2.3.x

   Specify a patch release number, if applicable:

## Product editions

Is this update specific to a product edition?

- [x] Magento Open Source only
- [x] Magento Commerce only

Is this update specific to an installed feature extension

- [ ] B2B extension
- [ ] Other feature set, please specify:

## Affected documentation pages

- https://docs.magento.com/user-guide/catalog/product-create-configurable.html
- https://docs.magento.com/user-guide/catalog/swatches.html
- https://docs.magento.com/user-guide/catalog/swatch-create.html

## Links to Magento source code or PRs

- https://jira.corp.magento.com/browse/MC-40719

### CODE REVIEW

File: src/catalog/product-create-configurable.md [hunk 1]
```diff
@@ -345,3 +345,7 @@ If you have a different image for each variation you can set the configuration t
 - The attribute set that is used as a template for a configurable product must include the attribute(s) that contain the values that are needed for each product variation.
 
 - The thumbnail image in the shopping cart can be set to display the image from the configurable product record or from the product variation.
+
+- Swatch attributes can be configured to not display corresponding simple product images when the swatch is selected by setting the "Update Product Preview Image" option value to "No" at the attribute edit page in the admin.
+
+- It is controlled by the theme how Image Gallery will behave when a user switches between product configurations. Default behavior for Blank theme is to override the parent configurable product images with selected product variation, while for Luma theme, the default behavior is to prepend selected product variation images to the parent configurable product images.
```
Comment (dobooth) [inline src/catalog/product-create-configurable.md:351]:
```suggestion
- The theme controls how the Image Gallery behaves when a user switches between product configurations. The default behavior for the Blank theme is to override the parent configurable product images with the selected product variation. For the Luma theme, the default behavior is to prepend the selected product variation images to the parent configurable product images.
```

File: src/catalog/swatch-create.md [hunk 2]
```diff
@@ -23,6 +23,8 @@ Use either of the following methods to create swatches for your products.
 
 1. Verify that **Catalog Input Type for Store Owner** is set to `Visual Swatch`.
 
+1. If it is preferred to not display corresponding simple product images when the swatch is selected on the product display page, set the "Update Product Preview Image" option value to "No".
```
Comment (dobooth) [inline src/catalog/swatch-create.md:26]:
```suggestion
1. If you prefer to not display corresponding simple product images when the swatch is selected on the product display page, set the "Update Product Preview Image" option value to "No".
```

File: src/catalog/swatch-create.md [hunk 3]
```diff
@@ -73,6 +75,8 @@ Use either of the following methods to create swatches for your products.
...
```
Comment (dobooth) [inline src/catalog/swatch-create.md:78]:
```suggestion
1. If you prefer to not display corresponding simple product images when the swatch is selected on the product display page, set the "Update Product Preview Image" option value to "No".
```

File: src/catalog/swatches.md [hunk 4]
```diff
@@ -6,6 +6,9 @@ Customers have high expectations when it comes to color, and it is crucial that
 
 For [configurable products]({% link catalog/product-create-configurable.md %}), color can be indicated by a visual swatch, text swatch, or input control. Swatches can be used on the product page, in product listings, and in [layered navigation]({% link catalog/navigation-layered.md %}). On the product page, swatches are synchronized to display the corresponding product image when the swatch is selected. When the customer selects the swatch, the corresponding value appears in the input field and the swatch is outlined as the current selection.
 
+{:.bs-callout-info}
+Swatch attributes can be configured to not display corresponding simple product images when the swatch is selected by setting the "Update Product Preview Image" option value to "No" at the attribute edit page in the admin.
```
Comment (dobooth) [inline src/catalog/swatches.md:10]:
```suggestion
Swatch attributes can be configured to not display corresponding simple product images when the swatch is selected by setting the "Update Product Preview Image" option value to "No" on the Attribute Edit page in the Admin.
```

**Review by dobooth — CHANGES_REQUESTED** [review]

**Review by odubovyk — APPROVED** [review]

**Review by dobooth — APPROVED** [review]

**Comment by dobooth** [comment]
running tests

**Pull request merged**